### PR TITLE
Add Sections in the PR Template

### PR DESCRIPTION
**Pull request by Future-Outlier**
# TL;DR
As title.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This change will improve the PR quality significantly.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/4366

## Related PRs
https://github.com/flyteorg/flyte/pull/4367

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]

**Comment by pingsutw** [comment]
cc @eapolinario @wild-endeavor wdyt

**Review by eapolinario — COMMENTED** [review]
Can we change comments to the style used in https://github.com/flyteorg/flyte/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L9-L13 ?

**Comment by Future-Outlier** [comment]
> Can we change comments to the style used in https://github.com/flyteorg/flyte/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1#L9-L13 ?

No problem!

File: pull_request_template.md [hunk 1]
```diff
@@ -1,26 +1,50 @@
-# TL;DR
-_Please replace this text with a description of what this PR accomplishes._
+<!--
+Thank you for sending the PR! 
+Please fill the applicable details below
+Happy contributing!
+-->
```
Comment (pingsutw) [inline pull_request_template.md:5]: Can we remove this

File: pull_request_template.md [hunk 2]
```diff
@@ -1,26 +1,50 @@
...
-https://github.com/flyteorg/flyte/issues/<number>
+## Docs link
 
-## Follow-up issue
-_NA_
-OR
-_https://github.com/flyteorg/flyte/issues/<number>_
+<!-- Add documentation link built by CI jobs here, and specify the changed place -->
+
+## Describe your changes
```
Comment (pingsutw) [inline pull_request_template.md:19]: And move this to the top

**Review by pingsutw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Review by pingsutw — APPROVED** [review]

**Pull request merged**